packages/toolkit/src/query/retry.ts

```diff
@@ -57,7 +68,12 @@ const retryWithBackoff: BaseQueryEnhancer<
     try {
       const result = await baseQuery(args, api, extraOptions)
       // baseQueries _should_ return an error property, so we should check for that and throw it to continue retrying
-      if (result.error) {
+      if (result.error && options.shouldRetry(result.error as FetchBaseQueryError, args, {
```

Comment:
If we are at `retry === options.maxRetries`, this will still be called. 

Yet, if the function returns `true`, no other retry will be made (see line 82). This is unexpected behavior for the user as they specifically signalled that they want to have a retry.

We have the following options:

1. if `retry === options.maxRetries`, do not call `shouldRetry` any more since it will not have any effect either way:
  this is the easiest solution
2. if `retry === options.maxRetries`, still call `shouldRetry`. If it returns `true`, somehow signal to the error handler that even though at that point, `retry > options.maxRetries` is `true`, another retry needs to be made.

Either option would need to be sufficiently documented. I am personally leaning towards 2., but it would complicate the code a bit.  
If you agree that 2. would be the preferable solution, maybe we could just have a default definition for `shouldRetry` to be `(_, __, {attempt, maxRetries}) => attempt < maxRetries` and swap out some logic around the retrying mechanism?

Reply:
I chose 2.
I've tried to implement it.
If the way it looks good, I will add a test at the end.